plugin/pkg/shoot/resourcereservation/admission.go

```diff
@@ -133,7 +142,7 @@ func (c *ResourceReservation) Admit(_ context.Context, a admission.Attributes, _
 		return nil
 	}
 
-	if !c.useGKEFormula {
+	if !c.useGKEFormula || !c.labelSelector.Matches(labels.Set(shoot.Labels)) {
```

Comment:
Good catch, this was not intended! I changed the default as suggested to match all Shoots in case no selector is specified. I didn't realize that after switching to `LabelSelectorAsSelector`, the default for an empty selector changed to `labels.Nothing()`. This wan't caught by the existing tests, as they still used `labels.Parse`.